node.go

```diff
@@ -62,9 +62,10 @@ func (n *Node) Save() error {
 	if err != nil {
 		return err
 	}
-	defer f.Close()
 
-	if err := json.NewEncoder(f).Encode(n); err != nil {
+	err = json.NewEncoder(f).Encode(n)
+	f.Close()
```

Comment:
err returned by `f.Close()` is ignored.  Should check/return if it fails for some reason.